src/btree/col_srch.c

```diff
@@ -254,6 +254,18 @@ __wt_col_search(
         if ((cip = __col_var_search(current, recno, NULL)) == NULL) {
             cbt->recno = __col_var_last_recno(current);
             cbt->slot = page->entries == 0 ? 0 : page->entries - 1;
+            /*
+             * We need to search the insert list in case there is nothing on the append list when we
+             * search it later on (which would be closer to the search record).
+             */
+            if (cbt->recno != WT_RECNO_OOB) {
+                ins_head = WT_COL_UPDATE_SLOT(page, cbt->slot);
+                ins = WT_SKIP_LAST(ins_head);
+                if (ins != NULL && cbt->recno == WT_INSERT_RECNO(ins)) {
+                    cbt->ins_head = ins_head;
+                    cbt->ins = ins;
+                }
```

Comment:
You also need analogous code in the FLCS branch above (basically the same thing, except WT_COL_UPDATE_SINGLE). Or, as I suggested below, move it down there where most of it can be shared.

Reply:
(That is, all the same logic applies so the bug doubtless affects FLCS too, it'll just be harder to detect there since it'll manifest as a wrong instead of missing value.)

Reply:
... below, above, depends on where you look at the comment I guess.